ExilenceNextApp/src/hooks/use-debounce.ts

```diff
@@ -0,0 +1,36 @@
+// credit to https://dev.to/gabe_ragland/debouncing-with-react-hooks-jci
+
+import { useEffect, useState } from 'react';
+
+// Our hook
+export function useDebounce(value: any, delay: number) {
```

Comment:
Maybe we could introduce [use-hooks package](https://github.com/dimitarnestorov/use-hooks#readme) with some more common and battle-tested approaches. WDYT?